includes/kaltura_derivatives.inc

```diff
@@ -68,10 +68,26 @@ function islandora_video_streaming_kaltura_upload($object, $force = false) {
 
       $entity = kaltura_entry_create(array('kaltura_entryid' => $entry->id));
       kaltura_map_properties($entry, $entity);
-
-      kaltura_save_entry_metadata($entity);
+      $entity->kaltura_title = $object->id; 
+      $entity->kaltura_description = $object->label;
+      
       kaltura_entry_save($entity); //write entity to drupal db
-
+    
+      try {
+        $kaltura_baseEntry = new KalturaBaseEntry();
+        $kaltura_baseEntry->name = $object->id;
+        $kaltura_baseEntry->description = $object->label;
+        $updated_kaltura = $client->baseEntry->update($entry->id, $kaltura_baseEntry);
+      } catch (Exception $e) {
+        $messages = array(
```

Comment:
This is spot-on for construction.  We also still have the problem of the [older code](https://github.com/pkp/browseBySection/commit/72d0ea62d21ce78211db875217c48a9c2e6ce6aa) blindly overwriting it, if both errors occur.  The larger `catch` is also a risk.  If code is added after this current `catch` but before the end of the `try`, and if that code throws an exception, our message is again overwritten by the broader `catch`.

By introducing this additional error handling, we need to fix the existing error handling.

Reply:
Rescind the 0's from each failureMessage statement?

Reply:
Consistently using the array append operator (`$array[]`) rather than the first index (`$array[0]`) should resolve this.  Note though, that you can't just remove the zero index.  That is, this won't work:
```
      $failureMessage['messages'][]['message'] = t('Failed to upload object @pid to Kaltura, exception: @exception');
      $failureMessage['messages'][]['message_sub']['@exception'] = $e->getMessage();
```

Another concern would be whether or not an warning would be triggered when first using the append operator on an undefined variable.  The first index might have been chosen because it didn't trigger such a warning.

Reply:
The procedure would be the same as the fix for Kaltura Base Entry exceptions, creating an array with the messages and then appending that to the failureMessage array?

I thought the failureMessages array was already defined earlier? If not, then a warning would not be triggered I think.

Reply:
Ah, yes.  We define it early with a "no exception caught" message.  So, if we continue to use that, we need to overwrite it in select cases (or better, probably just use it as a template?).

Reply:
I'm currently using the variable as a template in writing failure messages. 
This:
 ```
$failureMessage['messages'][0]['message'] = t('Failed to upload object @pid to Kaltura, exception: @exception');
      $failureMessage['messages'][0]['message_sub']['@exception'] = $e->getMessage();
```
To:
```
$messages = array(
          'message' => t('Failed to edit metadata in Kaltura'),
          'message_sub' => array('@pid' => $object->id),
          'type' => 'watchdog',
          'severity' => WATCHDOG_ERROR,
        );
        $failureMessage['messages'][] = $messages;
```

Reply:
Yes, `pid` should be added to the message substitutions.

Are you removing the [early definition of `$failureMessage` in this function](https://github.com/ulsdevteam/islandora_video_streaming/blob/38969904872a65be4ae7e99fd0393b369e8875e6/includes/kaltura_derivatives.inc#L38-L46)?  We don't want to always report a spurious error when a real error occurs.

Reply:
I do not think I am removing the early definition of $failureMessage, unless it is beneficial to the function.

Reply:
If `$failureMessage['messages']` is defined early, and you use `$failureMessage['messages'][]` to append a new message, you'll end up with both the early definition in `[0]` and the new message in `[1]`.